be/src/vec/functions/function_compress.cpp

```diff
@@ -0,0 +1,251 @@
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        uint32_t result, size_t input_rows_count) const override {
+        // Get the compression algorithm object
+        BlockCompressionCodec* compression_codec;
+        RETURN_IF_ERROR(get_block_compression_codec(segment_v2::CompressionTypePB::ZLIB,
+                                                    &compression_codec));
+
+        const auto& arg_column =
+                assert_cast<const ColumnString&>(*block.get_by_position(arguments[0]).column);
+
+        auto& arg_data = arg_column.get_chars();
+        auto& arg_offset = arg_column.get_offsets();
+        const char* arg_begin = reinterpret_cast<const char*>(arg_data.data());
+
+        auto result_column = ColumnString::create();
+        auto& col_data = result_column->get_chars();
+        auto& col_offset = result_column->get_offsets();
+        col_offset.resize(input_rows_count);
+
+        auto null_column = ColumnUInt8::create(input_rows_count);
+        auto& null_map = null_column->get_data();
+
+        std::string uncompressed;
+        Slice data;
+        Slice uncompressed_slice;
+
+        size_t total = arg_offset[input_rows_count - 1];
+        col_data.reserve(total * 1000);
+
+        for (size_t row = 0; row < input_rows_count; row++) {
+            null_map[row] = false;
+            data = Slice(arg_begin + arg_offset[row - 1], arg_offset[row] - arg_offset[row - 1]);
+            size_t data_length = arg_offset[row] - arg_offset[row - 1];
+
+            bool illegal = false;
+            // The first ten digits are "0x" and length, followed by hexadecimal, each two digits is a byte
+            if (data_length < 10 || data_length % 2 == 1) {
+                illegal = true;
+            } else {
+                if (data[0] != '0' || data[1] != 'x') {
+                    illegal = true;
+                }
+                for (size_t i = 2; i <= 9; i += 2) {
+                    if (!std::isxdigit(data[i])) {
+                        illegal = true;
+                    }
+                }
+            }
+
+            if (illegal) { // The top ten don't fit the rules
+                col_offset[row] = col_offset[row - 1];
+                // if data == "0x" return '', null_map[row] = false
+                if (!(data.size == 2 && data[0] == '0' && data[1] == 'x')) {
+                    null_map[row] = true;
+                }
+                continue;
+            }
+
+            unsigned int length = 0;
+            for (size_t i = 2; i <= 9; i += 2) {
+                unsigned char byte;
+                std::from_chars(data.data + i, data.data + i + 2, byte, 16);
+                length += (byte << (8 * (i / 2 - 1))); //Little Endian : 0x01000000 -> 1
+            }
+
+            uncompressed.resize(length);
+            uncompressed_slice = Slice(uncompressed);
```

Comment:
remove the `uncompressed`, just resize `col_data` to add its length with `length`. so just point `uncompressed_slice` into `col_data`. then we can do decompress inplace and no need to do heavily `memcpy`(at L237)